cms/static/js/views/utils/create_course_utils.js

```diff
@@ -1,57 +1,18 @@
 /**
  * Provides utilities for validating courses during creation, for both new courses and reruns.
  */
-define(["jquery", "underscore", "gettext", "js/views/utils/view_utils"],
-    function ($, _, gettext, ViewUtils) {
+define(["jquery", "gettext", "js/views/utils/view_utils", "js/views/utils/create_utils_base"],
+    function ($, gettext, ViewUtils, CreateUtilsFactory) {
+        "use strict";
         return function (selectors, classes) {
-            var validateTotalCourseItemsLength, setNewCourseFieldInErr, hasInvalidRequiredFields,
-                createCourse, validateFilledFields, configureHandlers;
+            CreateUtilsFactory.call(this, selectors, classes);
```

Comment:
They are returning a function. The function acts as a constructor. By using `call`, we use CreateUtilsFactory as an extension of the constructor. So, in reality, no prototyping is involved, we just steal all of that function's `this` assignments by passing it our own `this`. It's (almost) the equivalent of a PHP or C include, where all of the code would be dumped right into the current function. Not quite, because it wouldn't have access to all of the names, just `this`.